src/Progress/ProgressController/IProgressStepExecutionEvents.cs

```diff
@@ -18,6 +18,6 @@ public interface IProgressStepExecutionEvents
         /// </summary>
         /// <param name="progressDetailText">Optional (can be null)</param>
         /// <param name="progress">The execution progress</param>
-        void ProgressChanged(string progressDetailText, double progress);
+        void ProgressChanged(string progressDetailText, double progress = double.NaN);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use the overloading mechanism instead of the optional parameters. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS2360)